src/function_data.jl

```diff
@@ -18,6 +18,29 @@ LinearFunctionData(proportional_term) = LinearFunctionData(proportional_term, 0.
 get_proportional_term(fd::LinearFunctionData) = fd.proportional_term
 get_constant_term(fd::LinearFunctionData) = fd.constant_term
 
+function _transform_linear_vector_for_hdf(data::Vector{LinearFunctionData})
+    transfd_data = Vector{NTuple{2, Float64}}(undef, length(data))
+    for (ix, fd) in enumerate(data)
+        transfd_data[ix] = (get_proportional_term(fd), get_constant_term(fd))
+    end
+    return transfd_data
+end
+
+function transform_array_for_hdf(data::Vector{LinearFunctionData})
+    return transform_array_for_hdf(_transform_linear_vector_for_hdf(data))
+end
+
+function transform_array_for_hdf(
+    data::SortedDict{Dates.DateTime, Vector{LinearFunctionData}},
+)
```

Comment:
```
transform_array_for_hdf(SortedDict(k => _transform_linear_vector_for_hdf(v) for (k, v) in data])
```

Reply:
This won't allow for the use of `sizehint!`

Reply:
sizehint shouldn't be necessary if you build like this.